File: src/main/java/me/alegian/thaumcraft7/impl/init/registries/deferred/util/DeferredAspect.java
package me.alegian.thaumcraft7.impl.init.registries.deferred.util;

import me.alegian.thaumcraft7.impl.common.aspect.Aspect;
import net.minecraft.resources.ResourceKey;
import net.neoforged.neoforge.registries.DeferredHolder;

public class DeferredAspect<T extends Aspect> extends DeferredHolder<Aspect, T> {
  protected DeferredAspect(ResourceKey<Aspect> key) {
    super(key);
  }

  public static <T extends Aspect> DeferredAspect<T> createAspect(ResourceKey<Aspect> key) {
    return new DeferredAspect<>(key);
  }
}

File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/util/DeferredAspect.kt
package me.alegian.thavma.impl.init.registries.deferred.util

import me.alegian.thavma.impl.common.aspect.Aspect
import net.minecraft.resources.ResourceKey
import net.neoforged.neoforge.registries.DeferredHolder

class DeferredAspect<T : Aspect> private constructor(key: ResourceKey<Aspect>) :
    DeferredHolder<Aspect, T>(key) {
    companion object {
        fun <T : Aspect> createAspect(key: ResourceKey<Aspect>): DeferredAspect<T> {
            return DeferredAspect(key)
        }
    }
}
